garage/tf/algos/npo.py

```diff
@@ -33,14 +33,33 @@ class NPO(BatchPolopt):
     """
     Natural Policy Gradient Optimization.
 
-    Attributes:
-        name(str): The name of the algorithm.
+    Args:
+        pg_loss(enum): The type of loss functions to use.
         lr_clip_range(float): The limit on the likelihood ratio between
             policies, as in PPO.
         max_kl_step(float): The maximum KL divergence between old and new
             policies, as in TRPO.
+        optimizer(object): The optimizer of the algorithm.
+        optimizer_args(dict): The arguments of the optimizer.
+        name(str): The name of the algorithm.
         policy_ent_coeff(float): The coefficient of the policy entropy.
-        optimizer(float): The optimizer of the algorithm.
+        entropy_method(str): A string from: 'max', 'regularized'. The type
```

Comment:
Is adding entropy penalty to loss or reward enforced (instead of an option)?

Reply:
it's an option. you can also train with no entropy regularization.

also note that entropy is a bonus, not a penalty (we want more of it)

Reply:
should we add `None` check to `entropy_method`? my assumption is users can set `policy_ent_coeff=0.0` if they don't want to use entropy regularization

Reply:
hmm. you should do what you can to prevent the user from providing mutually-incompatible arguments.

e.g. a non-zero `policy_ent_coeff` when `entropy_method` is not `'no_entropy'` should be an error. Note that I recommend that the entropy method for none be `'no_entropy'` rather than the Python symbol `None`, because `None` represents that the user failed to make a choice or provide a value, *not* that the user has *chosen* to not use entropy.

Reply:
please add similar checks for other mutually-incompatible arguments, as you have noted in the comments.

only disallow arguments which make no sense together (e.g. guaranteed to fail or logically inconsistent). don't disallow argument combinations which seem useless or unlikely but not harmful.